app/View/Components/PostTile.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\View\Components;
+
+use Illuminate\View\Component;
+
+class PostTile extends Component
+{
+  public $class;
+  public $title;
+  public $link;
+  public $excerpt;
+  protected $postId;
+
+  /**
+   * Create a new component instance.
+   *
+   * @return void
+   */
+
+  public function __construct($class, $postId = null)
+  {
+    $this->class = $class;
+    $this->title = $this->getTitle();
+    $this->link = $this->getLink();
+    $this->excerpt = $this->getExcerpt();
+  }
+
+  protected function getTitle()
```

Comment:
I think that `getters` should return attribute values in this case. It seems to be ok if you would use `getTitle` function in the components, but in this case, getters are used only for setting attributes, so the meaning might be confusing. To I would use setters in constructors instead.

```php
public function __construct($class, $postId = null)
{
    $this->setTitle(get_the_title($this->postId));
}

private function setTitle(string $title): void
{
  $this->title = $title;
}
```